stat/roc_example_test.go

```diff
@@ -125,3 +125,41 @@ func ExampleROC_aUC() {
 	// false positive rate: [0 0.5 0.5 1 1]
 	// auc: 0.25
 }
+
+func ExampleTOC() {
+	classes := []bool{
+		false, true, false, false, true, false,
+		true, false, false, true, true, true,
+		true, true, true, true, false, false,
+		false, false, false, false, false, false,
+		false, false, false, false, false, false,
+	}
+
+	min, ntp, max := stat.TOC(classes, nil)
+	fmt.Printf("minimum bound: %v\n", min)
+	fmt.Printf("TOC:           %v\n", ntp)
+	fmt.Printf("maximum bound: %v\n", max)
+
+	// Output:
+	// minimum bound: [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 2 3 4 5 6 7 8 9 10]
+	// TOC:           [0 0 1 1 1 2 2 3 3 3 4 5 6 7 8 9 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10]
+	// maximum bound: [0 1 2 3 4 5 6 7 8 9 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10]
+}
+
+func ExampleTOC_unsorted() {
+	y := []float64{8, 7.5, 6, 5, 3, 0}
+	classes := []bool{true, true, true, false, true, false}
+	weights := []float64{2, 2, 3, 6, 1, 4}
+
+	stat.SortWeightedLabeled(y, classes, weights)
+
+	min, ntp, max := stat.TOC(classes, weights)
+	fmt.Printf("minimum bound: %v\n", min)
+	fmt.Printf("TOC:           %v\n", ntp)
+	fmt.Printf("maximum bound: %v\n", max)
+
+	// Output:
+	// minimum bound: [0 0 0 1 4 6 8]
+	// TOC:           [0 0 1 1 4 6 8]
+	// maximum bound: [0 4 5 8 8 8 8]
+}
```

Comment:
There is an issue here, you've sorted in increasing order of y, which makes this inconsistent with ROC and the documentation... we should clarify. If TOC is used in the fashsion shown in this example, then the interpretation would be consistent with the original version of ROC (where values LESS than or equal to a particular y cutoff are classified as "positive", this was later changed (parity reversed) to be consistent with sklearn so maybe we should be consistent to that. All you would need to do then is to specify that the classes should be ranked in _increasing_ order of the ranking variable y and if this is done then the interpretation of the output would be consistent with sklearn.

Reply:
Struggling with the wording for this. So pushing the logic changes.

Reply:
I'm struggling with the wording too, my brain is getting a bit mushy today I might need to come back to this another day if that's alright with you.

Reply:
SGTM

Reply:
Alright so I've thought about this abit more (sorry for the delay), and the issue is this: if there is some ranking variable y, and you sort the classes and weights in increasing order of y, this is in decreasing order of rank. I.e. the smallest value is y has the largest rank. So I think what we should do about it is clarify in the documentation. The issue originates from the phrase "where values at or above the given rank are assigned class true" because this is different to "where values at or above the given value of the ranking variable are assigned class true". One easy fix would be to just specify that we assign the smallest value in the ranking variable y rank "1", and the largest rank "len(y)". Then the interpretation as ">= a given rank" is the same as ">= a given value of the ranking variable". Does that make sense?

Reply:
Alternatively, we could leave it as is and just add the note that sorting in ascending order of the ranking variable is the same as in decending order of the rank variable. In which case that comment is fine.

Reply:
> One easy fix would be to just specify that we assign the smallest value in the ranking variable y rank "1", and the largest rank "len(y)".

This is the natural interpretation. I though that was known.

Reply:
Yeah, ok fair enough. It confused me though, but maybe that's because I'm over-thinking it.

Reply:
I would still spell it out for the benefit of simpletons like yours truly :) At least that the rank variable starts from 0 not from 1, since the convention in statistics seems to be to start from 1: https://en.wikipedia.org/wiki/Ranking#Ranking_in_statistics

Reply:
The zero'th element is not a ranked value.

Reply:
What do you mean?

Reply:
In the output slices, the zero'th element is not actually a ranked part, it's the origin. The ranked parts start at 1.

Reply:
Ah, OK.